daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Validation.scala

```diff
@@ -288,6 +288,21 @@ private final class Validation[Nid, Cid](implicit ECid: Equal[Cid]) {
                   keyIsReplayedBy(key1, key2) &&
                   result1 === result2 =>
               loop(rest1, rest2, stack)
+            case (
+                  Node.NodeRollback(
+                    children1,
+                    version1,
+                  ),
+                  Node.NodeRollback(
+                    children2,
+                    version2,
+                  ),
+                ) if version1 == version2 =>
+              loop(
+                children1.iterator.to(LazyList),
+                children2.iterator.to(LazyList),
+                stack,
+              )
```

Comment:
We have dropped `rest1` and `rest2` on the floor.
Don't we need to still process these?

Reply:
good point, I’ve pushed them to the stack like we do for exercise nodes 912b0f9e8e90b6f26ecfcc42455a1d5977d36016

Reply:
:+1: